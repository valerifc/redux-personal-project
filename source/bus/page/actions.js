// Types
import { types } from "./types";

export const pageActions = {
    setPage: (meta) => {
        return {
            type:    types.SET_PAGE,
            payload: meta,
        };
    },
};
