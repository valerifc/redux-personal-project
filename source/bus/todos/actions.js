// Types
import { types } from "./types";

export const todosActions = {
    // Sync
    fetchTodos: (todos) => {
        return {
            type:    types.FETCH_TODOS,
            payload: todos,
        };
    },
    createTodo: (todo) => {
        return {
            type:    types.CREATE_TODO,
            payload: todo,
        };
    },
    updateTodo: (todo) => {
        return {
            type:    types.UPDATE_TODO,
            payload: todo,
        };
    },
    updateTodos: (todos) => {
        return {
            type:    types.UPDATE_TODOS,
            payload: todos,
        };
    },
    updateAllTodos: (todo) => {
        return {
            type:    types.UPDATE_ALL_TODOS,
            payload: todo,
        };
    },
    removeTodo: (todoId) => {
        return {
            type:    types.REMOVE_TODO,
            payload: todoId,
        };
    },

    // Async
    fetchTodosAsync: (queryParams) => {
        return {
            type:    types.FETCH_TODOS_ASYNC,
            payload: queryParams,
        };
    },
    createTodoAsync: (message) => {
        return {
            type:    types.CREATE_TODO_ASYNC,
            payload: message,
        };
    },
    updateTodoAsync: (todo) => {
        return {
            type:    types.UPDATE_TODO_ASYNC,
            payload: todo,
        };
    },
    removeTodoAsync: (todoId) => {
        return {
            type:    types.REMOVE_TODO_ASYNC,
            payload: todoId,
        };
    },
};
