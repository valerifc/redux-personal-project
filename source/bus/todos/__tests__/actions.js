// Actions
import { todosActions } from "../actions";

describe('todos actions:', () => {
    test('fetchTodos', () => {
        expect(todosActions.fetchTodos(__.todos)).toMatchSnapshot();
    });

    test('createTodo', () => {
        expect(todosActions.createTodo(__.todo)).toMatchSnapshot();
    });

    test('updateTodo', () => {
        expect(todosActions.updateTodo(__.todo)).toMatchSnapshot();
    });

    test('updateTodos', () => {
        expect(todosActions.updateTodos(__.todos)).toMatchSnapshot();
    });

    test('updateAllTodos', () => {
        expect(todosActions.updateAllTodos(__.todo)).toMatchSnapshot();
    });

    test('removeTodo', () => {
        expect(todosActions.removeTodo(__.todoId)).toMatchSnapshot();
    });

    test('fetchTodosAsync', () => {
        expect(todosActions.fetchTodosAsync(__.queryParams)).toMatchSnapshot();
    });

    test('createTodoAsync', () => {
        expect(todosActions.createTodoAsync(__.message)).toMatchSnapshot();
    });

    test('updateTodoAsync', () => {
        expect(todosActions.updateTodoAsync(__.todo)).toMatchSnapshot();
    });

    test('removeTodoAsync', () => {
        expect(todosActions.removeTodoAsync(__.todoId)).toMatchSnapshot();
    });
});
