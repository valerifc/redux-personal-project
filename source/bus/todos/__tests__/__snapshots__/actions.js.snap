// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todos actions: createTodo 1`] = `
Object {
  "payload": Object {
    "completed": false,
    "created": "2018-02-10T15:44:32.959Z",
    "favorite": false,
    "id": "5a7f136031a5d90001271635",
    "message": "Hello",
  },
  "type": "CREATE_TODO",
}
`;

exports[`todos actions: createTodoAsync 1`] = `
Object {
  "payload": "TEST_MESSAGE",
  "type": "CREATE_TODO_ASYNC",
}
`;

exports[`todos actions: fetchTodos 1`] = `
Object {
  "payload": Array [
    Object {
      "completed": true,
      "created": "2018-02-10T15:44:34.624Z",
      "favorite": false,
      "id": "5a7f136231a5d90001271637",
      "message": "Hello Andrey!",
      "modified": "2018-02-10T16:01:12.406Z",
    },
    Object {
      "completed": false,
      "created": "2018-02-10T15:44:33.675Z",
      "favorite": false,
      "id": "5a7f136131a5d90001271636",
      "message": "Hello",
    },
    Object {
      "completed": false,
      "created": "2018-02-10T15:44:32.959Z",
      "favorite": false,
      "id": "5a7f136031a5d90001271635",
      "message": "Hello",
    },
  ],
  "type": "FETCH_TODOS",
}
`;

exports[`todos actions: fetchTodosAsync 1`] = `
Object {
  "payload": Object {
    "page": 2,
    "search": "test_search",
    "size": 10,
  },
  "type": "FETCH_TODOS_ASYNC",
}
`;

exports[`todos actions: removeTodo 1`] = `
Object {
  "payload": "5a7f136231a5d90001271637",
  "type": "REMOVE_TODO",
}
`;

exports[`todos actions: removeTodoAsync 1`] = `
Object {
  "payload": "5a7f136231a5d90001271637",
  "type": "REMOVE_TODO_ASYNC",
}
`;

exports[`todos actions: updateAllTodos 1`] = `
Object {
  "payload": Object {
    "completed": false,
    "created": "2018-02-10T15:44:32.959Z",
    "favorite": false,
    "id": "5a7f136031a5d90001271635",
    "message": "Hello",
  },
  "type": "UPDATE_ALL_TODOS",
}
`;

exports[`todos actions: updateTodo 1`] = `
Object {
  "payload": Object {
    "completed": false,
    "created": "2018-02-10T15:44:32.959Z",
    "favorite": false,
    "id": "5a7f136031a5d90001271635",
    "message": "Hello",
  },
  "type": "UPDATE_TODO",
}
`;

exports[`todos actions: updateTodoAsync 1`] = `
Object {
  "payload": Object {
    "completed": false,
    "created": "2018-02-10T15:44:32.959Z",
    "favorite": false,
    "id": "5a7f136031a5d90001271635",
    "message": "Hello",
  },
  "type": "UPDATE_TODO_ASYNC",
}
`;

exports[`todos actions: updateTodos 1`] = `
Object {
  "payload": Array [
    Object {
      "completed": true,
      "created": "2018-02-10T15:44:34.624Z",
      "favorite": false,
      "id": "5a7f136231a5d90001271637",
      "message": "Hello Andrey!",
      "modified": "2018-02-10T16:01:12.406Z",
    },
    Object {
      "completed": false,
      "created": "2018-02-10T15:44:33.675Z",
      "favorite": false,
      "id": "5a7f136131a5d90001271636",
      "message": "Hello",
    },
    Object {
      "completed": false,
      "created": "2018-02-10T15:44:32.959Z",
      "favorite": false,
      "id": "5a7f136031a5d90001271635",
      "message": "Hello",
    },
  ],
  "type": "UPDATE_TODOS",
}
`;
